src/lib/session.ts

```diff
@@ -0,0 +1,80 @@
+import { derived, get, writable } from "svelte/store";
+
+interface Session {
+  id: string;
+  publicKey: string;
+}
+const store = writable<Session | undefined>(undefined);
+export const sessionStore = derived(store, s => s);
+
+export async function authenticate(params: {
+  id: string;
+  signature: string;
+  publicKey: string;
+}): Promise<"success" | "failure"> {
+  disconnect();
+
+  const request = await fetch(
+    `http://0.0.0.0:8080/api/v1/sessions/${params.id}`,
```

Comment:
I think for this MVP this is ok, but there is the possibility that users run their node with other ports.
Maybe we should allow at some point to add a input field to the Connect tooltip for custom ports?

Reply:
Fair. But I think we should wait for people to request this before we make it customisable.